gui/src/actions.js

```diff
@@ -316,3 +250,105 @@ export const updateBridge = (data, successCallback, errorCallback) => {
       })
   }
 }
+
+// DEV NOTE:
+// Above here is deprecated. Use the `request(...)` function below to wrap API
+// calls in a counter, normalize JSON-API responses and create notifications.
+//
+// The calls above will be converted gradually.
+const handleError = dispatch => error => {
+  if (error instanceof AuthenticationError) {
+    dispatch(redirectToSignOut())
+  } else {
+    dispatch(notifyError(({msg}) => msg, error))
+  }
+}
+
+const request = (type, requestData, normalizeData, ...apiArgs) => {
+  return dispatch => {
+    dispatch({type: `REQUEST_${type}`})
+    return requestData(...apiArgs)
+      .then(json => {
+        const data = normalizeData(json)
+        dispatch({type: `UPSERT_${type}`, data: data})
+      })
+      .catch(handleError(dispatch))
+      .finally(() => dispatch({type: `RESPONSE_${type}`}))
+  }
+}
+
+const normalizeFetchJob = json => {
+  const attrs = Object.assign({}, json.data.attributes)
+  const runs = attrs.runs
+  delete attrs.runs
+
+  const validJsonApi = Object.assign(
+    {},
+    json,
+    {
+      data: {
+        id: json.data.id,
+        type: json.data.type,
+        attributes: attrs,
```

Comment:
Optional: Could we collapse this to just be 
```js
{ data: Object.assign(json.data, { relationships: ...}) }
```

? You delete runs on a reference above so it will mold the original.